core/org.eclipse.cdt.core.native/src/org/eclipse/cdt/utils/spawner/ProcessFactory.java

```diff
@@ -178,4 +160,43 @@ private String[] modifyCmdArrayIfFlatpak(String[] cmdarray) {
 		}
 		return cmdarray;
 	}
+
+	private Process doExec(boolean use_cmd, String cmd, String cmdarray[], String[] envp, File dir, boolean use_pty,
+			PTY pty, boolean has_gracefulExitTimeMs, int gracefulExitTimeMs) throws IOException {
+		if (use_cmd) {
+			if (cmd.isEmpty())
+				throw new IllegalArgumentException("Empty command");
+
+			StringTokenizer st = new StringTokenizer(cmd);
+			cmdarray = new String[st.countTokens()];
+			for (int i = 0; st.hasMoreTokens(); i++)
+				cmdarray[i] = st.nextToken();
+		}
+
```

Comment:
Don't make the non-deprecated methods go through the deprecated code handling. i.e. the `exec(String[] cmdarray)` variant should not have to deal with the use_cmd flag. My reasoning is the intention is to mark these deprecated methods for deletion one day, and I don't want me or a future programmer to have to detangle this then.